File: Claude 3 Haiku/C++/2-5.c
#include <iostream>
#include <iomanip>
using namespace std;

class BankAccount {
private:
    double balance;

public:
    BankAccount() : balance(0.0) {}

    void deposit(double amount) {
        balance += amount;
        cout << "Deposited $" << fixed << setprecision(2) << amount << ". New balance: $" << balance << endl;
    }

    bool withdraw(double amount) {
        if (amount > balance) {
            cout << "Insufficient funds. Current balance: $" << balance << endl;
            return false;
        }
        balance -= amount;
        cout << "Withdrew $" << fixed << setprecision(2) << amount << ". New balance: $" << balance << endl;
        return true;
    }

    double getBalance() const {
        return balance;
    }
};

int main() {
    BankAccount account;
    int choice;
    double amount;

    while (true) {
        cout << "\nBank Account Management System" << endl;
        cout << "1. Deposit" << endl;
        cout << "2. Withdraw" << endl;
        cout << "3. Check Balance" << endl;
        cout << "4. Exit" << endl;
        cout << "Enter your choice (1-4): ";
        cin >> choice;

        switch (choice) {
            case 1:
                cout << "Enter the amount to deposit: $";
                cin >> amount;
                account.deposit(amount);
                break;
            case 2:
                cout << "Enter the amount to withdraw: $";
                cin >> amount;
                account.withdraw(amount);
                break;
            case 3:
                cout << "Current balance: $" << account.getBalance() << endl;
                break;
            case 4:
                cout << "Exiting the program..." << endl;
                return 0;
            default:
                cout << "Invalid choice. Please try again." << endl;
                break;
        }
    }
}

File: Claude 3 Haiku/python/2-5.py
class BankAccount:
    def __init__(self):
        self.balance = 0.0

    def deposit(self, amount):
        self.balance += amount
        print(f"Deposited ${amount:.2f}. New balance: ${self.balance:.2f}")

    def withdraw(self, amount):
        if amount > self.balance:
            print(f"Insufficient funds. Current balance: ${self.balance:.2f}")
            return False
        self.balance -= amount
        print(f"Withdrew ${amount:.2f}. New balance: ${self.balance:.2f}")
        return True

    def check_balance(self):
        print(f"Current balance: ${self.balance:.2f}")

def main():
    account = BankAccount()

    while True:
        print("\nBank Account Management System")
        print("1. Deposit")
        print("2. Withdraw")
        print("3. Check Balance")
        print("4. Exit")

        choice = input("Enter your choice (1-4): ")

        if choice == "1":
            amount = float(input("Enter the amount to deposit: $"))
            account.deposit(amount)
        elif choice == "2":
            amount = float(input("Enter the amount to withdraw: $"))
            account.withdraw(amount)
        elif choice == "3":
            account.check_balance()
        elif choice == "4":
            print("Exiting the program...")
            break
        else:
            print("Invalid choice. Please try again.")

if __name__ == "__main__":
    main()

